src/contrib/cluster/Akka.Cluster.Sharding.Tests/SupervisionSpec.cs

```diff
@@ -133,9 +133,18 @@ public void SupervisionSpec_for_a_sharded_actor_must_allow_passivation()
             region.Tell(new Msg(10, "passivate"));
             ExpectTerminated(response.Self);
 
-            // This would fail before as sharded actor would be stuck passivating
-            region.Tell(new Msg(10, "hello"));
-            ExpectMsg<Response>(TimeSpan.FromSeconds(20));
+            Within(TimeSpan.FromSeconds(10), () =>
+            {
+                // This would fail before as sharded actor would be stuck passivating
+                // Need to use AwaitAssert here - message can be sent the the BackOffSupervisor, which is now
+                // terminating but still alive, but its child (the ultimate recipient of the message) is dead
+                // and this message will go to DeadLetters.
+                AwaitAssert(() =>
```

Comment:
cc @ismaelhamed - saw this test failing frequently and the logs revealed the following:

```
[INFO][1/22/20 5:59:58 PM][Thread 0063][Cluster (akka://test)] Cluster Node [akka.tcp://test@0.0.0.0:43337] - Node [akka.tcp://test@0.0.0.0:43337] is JOINING itself (with roles []) and forming a new cluster
[INFO][1/22/20 5:59:58 PM][Thread 0063][Cluster (akka://test)] Cluster Node [akka.tcp://test@0.0.0.0:43337] - Leader is moving node [akka.tcp://test@0.0.0.0:43337] to [Up]
[INFO][1/22/20 5:59:58 PM][Thread 0005][akka.tcp://test@0.0.0.0:43337/system/sharding/passyCoordinator] Singleton manager started singleton actor [akka://test/system/sharding/passyCoordinator/singleton] 
[INFO][1/22/20 5:59:58 PM][Thread 0005][akka.tcp://test@0.0.0.0:43337/system/sharding/passyCoordinator] ClusterSingletonManager state change [Start -> Oldest] Akka.Cluster.Tools.Singleton.Uninitialized
[INFO][1/22/20 5:59:58 PM][Thread 0064][akka.tcp://test@0.0.0.0:43337/system/sharding/passy] passy: Idle entities will be passivated after [00:02:00]
[INFO][1/22/20 6:00:00 PM][Thread 0058][akka.tcp://test@0.0.0.0:43337/system/sharding/passy/0/10/child] Starting
[INFO][1/22/20 6:00:00 PM][Thread 0058][akka.tcp://test@0.0.0.0:43337/system/sharding/passy/0/10/child] Passivating
[INFO][1/22/20 6:00:00 PM][Thread 0058][akka.tcp://test@0.0.0.0:43337/system/sharding/passy/0/10/child] Stopping
[INFO][1/22/20 6:00:00 PM][Thread 0063][akka://test/system/sharding/passy/0/10/child] Message [String] from akka://test/system/testActor52 to akka://test/system/sharding/passy/0/10/child was not delivered. [1] dead letters encountered .This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
[INFO][1/22/20 6:00:00 PM][Thread 0005][akka://test/deadLetters] Message [StopMessage] from akka://test/system/sharding/passy/0 to akka://test/deadLetters was not delivered. [2] dead letters encountered .This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
[DEBUG][1/22/20 6:00:20 PM][Thread 0003][ActorSystem(test)] System shutdown initiated
[INFO][1/22/20 6:00:20 PM][Thread 0127][/system/remoting-terminator] Shutting down remote daemon.
[INFO][1/22/20 6:00:20 PM][Thread 0127][/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[INFO][1/22/20 6:00:20 PM][Thread 0063][remoting (akka://test)] Remoting shut down
[INFO][1/22/20 6:00:20 PM][Thread 0127][/system/remoting-terminator] Remoting shut down.
[DEBUG][1/22/20 6:00:20 PM][Thread 0015][EventStream] Shutting down: StandardOutLogger started
[DEBUG][1/22/20 6:00:20 PM][Thread 0015][EventStream] All default loggers stopped
```

This is because, as I explain the comment, I believe that the `BackoffSupervisor` can still be alive (we death watch the child) thus the `Shard` sends the message to the old `IActorRef` even though it's in the midst of shutting down. I rewrote this to use `AwaitAssert` so we can retry in the event we lose that race.